storage/rocksdb/ha_rocksdb.cc

```diff
@@ -5216,8 +5216,17 @@ int ha_rocksdb::load_hidden_pk_value() {
   active_index = m_tbl_def->m_key_count - 1;
   const uint8 save_table_status = table->status;
 
-  Rdb_transaction *const tx = get_or_create_tx(table->in_use);
-  const bool is_new_snapshot = !tx->has_snapshot();
+  /*
+    We should read the latest committed value in the database.
```

Comment:
Is this needed, or is this being extra defensive, but not strictly required? It seems the fix needed is transferring m_hidden_pk_val from the old table_def to the new table_def during alter, and that is already in this diff. This code only populates the m_hidden_pk_val of the table if it hasn't been loaded yet. If there is an open transaction with the table, then we shouldn't be able to rename the table. If the current snapshot does not read the latest data, then another transaction must have written the new m_hidden_pk_val after this snapshot was created. In that case, the table definition should have the latest data, and whether we read with the latest snapshot or the existing snapshot shouldn't matter, no?

Reply:
Also discovered this with valgrind: if one calls `tx->start_tx`, they must call set_params beforehand, like it's done at other places:
```
    tx->set_params(THDVAR(thd, lock_wait_timeout), THDVAR(thd, max_row_locks));
    tx->start_tx();
```

Reply:
... as discussed on the call - the patch is extra defensive, and I will remove this part.  Will leave only the one in `Rdb_ddl_manager::rename`.